core/lib/resources/presence/index.js

```diff
@@ -33,7 +33,7 @@ Presence.prototype.setup = function() {
   var self = this;
 
   this.manager = new PresenceManager(this.name, this.options.policy, Presence.sentry);
-  this.manager.on('user_online', function(userId, userType, userData) {
+  this.manager.on('user_online', function(userId, userType, userData, data) {
```

Comment:
suggestion: Branch off lflorio/presence_states. Fix all this long parameter problems. Create a new PR into lflorio/presence_states. Get a +1 on this PR and that PR. Then merge correctly.